src/sigchain/sigchain.iced

```diff
@@ -124,21 +156,26 @@ class ChainLink
       err = new E.FingerprintMismatchError "signing fingerprint (#{signing_fingerprint}) and payload fingerprint (#{payload_fingerprint}) mismatch"
     cb err
 
-  @_check_reverse_signatures : ({payload, parsed_keys}, cb) ->
+  @_check_reverse_signatures : ({payload, key_state}, cb) ->
     esc = make_esc cb, "ChainLink._check_reverse_signatures"
     if payload.body.sibkey?
       kid = payload.body.sibkey.kid
-      key_manager = parsed_keys.get_key_manager kid
-      if not key_manager?
+      full_hash = payload.body.sibkey.full_hash
+      sibkey_key_manager = if full_hash?
+        # key_state hasn't been updated with the new sibkey's full hash yet
+        key_state.parsed_keys.get_pgp_key_manager_with_hash kid, full_hash
```

Comment:
Can we add a `TODO` here? In theory we should be testing the case where a merged version of the sibkey was valid, but the full hash version at the time of delegation technically was not.